### DISCUSSION TITLE
IDE questions

### DISCUSSION THREAD

**Discussion by dmpetrov** [treeverse/dvc#5915] [New Features & Ideas]
Do we need to integrate DVC to IDEs?

Details: https://discuss.dataversioncontrol.com/t/ide-integration-or-online-service/22

**Comment by prihoda** [comment]
Hi guys, are you considering the IDE integrations? 

I think a PyCharm integration would be very useful, starting with basic support for DVC files, like clicking on file paths to open the file, automatic status of md5 checksums, adding DVC reproduction command to the context menu, etc.

**Comment by efiop** [comment]
Hi @prihoda !

We are thinking about integrations for different IDEs, but it is not a top priority right now :slightly_frowning_face: We would be extremely happy to help with anything we can, if anyone is willing to create such plugin.

**Comment by prihoda** [comment]
Hey guys, 

I gave it a shot :) Check it out: https://plugins.jetbrains.com/plugin/11368-dvc-support-poc

The code is here: https://github.com/prihoda/intellij-dvc-support-poc

![image](https://user-images.githubusercontent.com/2894124/49166825-6d283280-f334-11e8-9a4a-5d3d2ee2781c.png)

Current features:
- Bash syntax highlighting in cmd field (when BashSupport plugin is installed)
- Click-through file references for dependencies and outputs (ctrl+click)
- DVC file syntax highlighting (YAML)
- DVC file icon

If you like, I'll be happy to move the repository under your command and give it a nicer name in the plugin repository. Then, we should be able to suggest this plugin when a DVC file is open, like so:

![image](https://user-images.githubusercontent.com/2894124/48985151-2accde80-f105-11e8-9b2f-59447ef3f729.png)

Now the question is, where do we want to take this :) What do you see as the next major features? Is anyone of you actively using PyCharm? Would you be considering full integration as with GIT or are you looking for just a "read only" approach? It would be a lot of work, especially due to compatibility with different DVC versions, but in my opinion it would definitely be worth it. I would really appreciate at least real-time status of all DVC files shown using the DVC file icon.

I'd be happy to meet online and discuss how I can help!

**Comment by ghost** [comment]
Nice, @prihoda!

We have our Discord channel where we discuss ideas with the community and help them with questions they have about DVC: https://dvc.org/chat

Feel free to join and ping us!

> Is anyone of you actively using PyCharm?

I'm sadly not a PyCharm user :disappointed: 

> ... especially due to compatibility with different DVC versions, but in my opinion it would definitely be worth it

I hope that after releasing version `1.0.0` the API would not have any breaking changes (until `2.0.0`), so it would be easier to integrate DVC with other tools.

**Comment by efiop** [comment]
Hi @prihoda !

Wow! That is amazing! :tada: I'm not a PyCharm user myself, but having this plugin honestly makes me want to switch to it :slightly_smiling_face: How about we schedule a hangouts call this week to discuss this? We've also wanted to meet you and ask you some questions for the longest time :slightly_frowning_face: If you are interested, please shoot me an email(see in my profile), so we could schedule something. 

Thanks,
Ruslan

**Comment by shcheklein** [comment]
This awesome! Thanks a lot. Let's have a chat like @efiop suggested to discuss a name, place for the repository, future plans. I would def love to see a deeper integration! I'm jetbrains products fan, @dmpetrov and our advisor Vova (top-20 Kaggle at some point) are among users.
